sympy/functions/elementary/trigonometric.py

```diff
@@ -1948,6 +1948,26 @@ def eval(cls, arg):
         if i_coeff is not None:
             return S.ImaginaryUnit * asinh(i_coeff)
 
+        if isinstance(arg, sin):
+            ang = arg.args[0]
+            if not ang.has(Symbol) and ang.is_real:
```

Comment:
Here and in other places, I think `if ang.is_comparable:` would be a better check, replacing both parts of this "and" clause. A SymPy expression may fail to be numeric without containing any Symbol objects. For example, try
```
f = Function('f', real=True)
asin(sin(f(3)))
```
and you'll get an error. The property  `is_comparable` is what needs to be true for relations like `ang > 2*pi` to be evaluated to a boolean. Reference: 
https://github.com/sympy/sympy/blob/master/sympy/core/basic.py#L572

Reply:
Thanks, for the suggestion. 
Indeed it would be a better check. 
Committed to that.